### PR TITLE
Add the context menu component

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**
 * Context menu is a clickable ellipsis
 * When not open only the ellipsis is shown
 * When open a menu of actions is shown
 * If a user clicks the action in the menu the corresponding aciton is called
 * The menu closes 'onBlur'

**How to test**
Verify tests pass

**Screenshots**

Screenshot from 2021-02-15 15-39-45
---
Screenshot from 2021-02-15 15-39-57

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/296

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Review by gopar — APPROVED** [review]
Awesome! I was totally doing the wrong thing when trying to build it.

File: frontend/src/components/ContextMenu.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  };
+
+  return (
+    <div onBlur={onBlur}>
+      <Button
+        type="button"
+        className="smart-hub--context-menu-button smart-hub--button__no-margin"
+        unstyled
+        onClick={() => updateShown((previous) => !previous)}
+        aria-label={label}
```
Comment (rahearn) [inline frontend/src/components/ContextMenu.js:40]: Good use of the `aria-label` but I think for accessibility there's likely some more we need to do. [`aria-haspopup`](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20161214/examples/menu-button/menu-button-1/menu-button-1.html) might be a good thing to look at

**Review by rahearn — COMMENTED** [review]
Do you have a quick gist or code sample for where I can plug the `<ContextMenu>` in to manually test it out?

File: frontend/src/components/ContextMenu.js (hunk 1 above)
Comment (rahearn) [reply frontend/src/components/ContextMenu.js:40]: Here's probably a better link for `aria-haspopup` https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup

**Review by rahearn — CHANGES_REQUESTED** [review]
One more accessibility issue. When using VoiceOver, it announces that the popup menu can be closed with `Escape` but that doesn't work.

Beyond that, there were some styling issues (and usage issues, since the button's active region didn't perfectly line up with the visual region, leading to activated clicks when I thought I'd be closing the menu) when I tried to attach the context menu to the activity reports list. I'm ok with not addressing those until this component is in use in those places, but I wanted to note that they'd be something to watch out for.

**Comment by jasalisbury** [comment]
> Do you have a quick gist or code sample for where I can plug the `<ContextMenu>` in to manually test it out?

@rahearn I could have sworn I sent an example on how to use the `<ContextMenu>` but I can't find it anywhere... Did I actually send you an example somewhere?

**Comment by rahearn** [comment]
> Did I actually send you an example somewhere?

No, but that's ok, it's not the most complicated component to wire up. I used it this way:

```
diff --git a/frontend/src/pages/Landing/index.js b/frontend/src/pages/Landing/index.js
index 3152da2..c1fbed5 100644
--- a/frontend/src/pages/Landing/index.js
+++ b/frontend/src/pages/Landing/index.js
@@ -132,9 +132,7 @@ function renderReports(reports) {
           </Tag>
         </td>
         <td>
-          <button type="button" className="smart-hub--dotdotdot">
-            ...
-          </button>
+          <ContextMenu label="options" menuItems={[{label: "Edit", onClick: () => { console.log("Edit")}}, {label: "Delete", onClick: () => { console.log("Delete") }}]} />
         </td>
       </tr>
     );
```

**Comment by jasalisbury** [comment]
> Beyond that, there were some styling issues (and usage issues, since the button's active region didn't perfectly line up with the visual region, leading to activated clicks when I thought I'd be closing the menu) when I tried to attach the context menu to the activity reports list. I'm ok with not addressing those until this component is in use in those places, but I wanted to note that they'd be something to watch out for.

Are you talking about the buttons being the full width of the menu? If so that was intentional, but as you say we can address it when the component is in use.

**Comment by rahearn** [comment]
> Are you talking about the buttons being the full width of the menu?

This is what I'm seeing on Chrome with the context menu on the activity reports table. The focus outline is where I can click on the button, but you can see it doesn't extend to the visual left, and it's beyond the visual right. Maybe a bad interaction with the `SimpleBar` components?

Screen Shot 2021-02-17 at 8 49 21 PM

**Review by rahearn — APPROVED** [review]

**Comment by jasalisbury** [comment]
> > Are you talking about the buttons being the full width of the menu?
> 
> This is what I'm seeing on Chrome with the context menu on the activity reports table. The focus outline is where I can click on the button, but you can see it doesn't extend to the visual left, and it's beyond the visual right. Maybe a bad interaction with the `SimpleBar` components?
> 
> Screen Shot 2021-02-17 at 8 49 21 PM

Oh yeah I don't see that where I put it, we'll probably have to mess with the styles when it gets placed in the table.

**Pull request merged**